pkg/accounts/kubernetes/validator.go

```diff
@@ -122,33 +131,44 @@ func makeClientFromSecretRef(ctx context.Context, ref *interfaces.SecretInNamesp
 	if err != nil {
 		return nil, err
 	}
-	return config.ClientConfig()
+
+	cfg, err := config.RawConfig()
+	if err != nil {
+		return nil, fmt.Errorf("Error parsing kubeconfigFile:\n  %w\n  kubeconfigFile contents:\n\n%s", err, str)
+	}
+	if settings.Context != "" {
+		cfg.CurrentContext = settings.Context
+	}
+	return clientcmd.NewDefaultClientConfig(cfg, &clientcmd.ConfigOverrides{}).ClientConfig()
 }
 
 // makeClientFromConfigAPI makes a client config from the v1 Config (the usual format for kubeconfig) inlined
 // into the CRD.
-func makeClientFromConfigAPI(config *clientcmdv1.Config) (*rest.Config, error) {
+func makeClientFromConfigAPI(config *clientcmdv1.Config, settings *authSettings) (*rest.Config, error) {
 	cfg := clientcmdapi.NewConfig()
 	if err := clientcmdlatest.Scheme.Convert(config, cfg, nil); err != nil {
 		return nil, nil
 	}
+	if settings.Context != "" {
```

Comment:
If settings cannot be nil, don't pass as pointer.

As a side note, I understand why we'd support `context` here for consistency but in effect the user would be defining a kubeconfig with multiple contexts but only one is used (in a secret it makes sense since the kubeconfig can be shared):

```
kind: SpinnakerAccount
spec:
  type: Kubernetes
  kubernetes:
    kubeconfig:
      <inlined kubeconfig>
    settings:
      context: the-context-i-want  
```